src/main/java/io/jenkins/update_center/HPI.java

```diff
@@ -189,8 +194,8 @@ public String getDescription() throws IOException {
             ArtifactCoordinates coordinates = new ArtifactCoordinates(artifact.groupId, artifact.artifactId, artifact.version, "jar");
             try (InputStream is = repository.getZipFileEntry(new MavenArtifact(repository, coordinates), "index.jelly")) {
                 StringBuilder b = new StringBuilder();
-                HtmlStreamRenderer renderer = HtmlStreamRenderer.create(b, Throwable::printStackTrace, html -> LOGGER.log(Level.INFO, "Bad HTML: '" + html + "' in " + artifact.getGav()));
-                HtmlSanitizer.sanitize(IOUtils.toString(is, StandardCharsets.UTF_8), HTML_POLICY.apply(renderer), PRE_PROCESSOR);
+                @NonNull HtmlStreamRenderer renderer = HtmlStreamRenderer.create(b, Throwable::printStackTrace, html -> LOGGER.log(Level.INFO, "Bad HTML: '" + html + "' in " + artifact.getGav()));
```

Comment:
This looks weird, what's happening here?

Reply:
SInce `apply` on Guava's `Function` is considered nullable and `PolicyFactory` does not override that annotation, SpotBugs sees this as a potential NPE source. The two ways to fix it seem to be 1) extracting to a method *without return type annotation* or 2) changing the declaring type to `java.util.function.Function`. I changed the solution from 1) to 2) and added comment.